### PR TITLE
IDEMPIERE-5754 - Check PInstance_ID before Process Lock DB Update, PInstance 0 on createPricelist

### PR DESCRIPTION
**Pull request by PeterTakacs300**
Applying changes from @igorpojzl.

JIRA ticket: https://idempiere.atlassian.net/browse/IDEMPIERE-5754

# Pull Request Checklist

- [x] My code follows the [code guidelines](https://wiki.idempiere.org/en/Contributing_to_Trunk) of this project
- [x] My code follows the best practices of this project
- [x] I have performed a self-review of my own code
- [x] My code is easy to understand and review. 
- [x] In hard-to-understand areas, I have commented my code.
- [x] My changes generate no new warnings
- [ ] New and existing unit tests pass locally with my changes
- [ ] I have added unit tests that prove my fix is effective or that my feature works
- [x] Any dependent changes have been merged and published in downstream modules
- [x] I have checked my code and corrected any misspellings
- [ ] I have made corresponding changes to the documentation as follows:
  - [ ] New feature (non-breaking change which adds functionality): I have created the New Feature page in the project wiki explaining the functionality and how to use it. If relevant, I have committed sample data to the core seed to have usable examples in GardenWorld.
  - [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected): I have documented the change in a clear way that everyone in the community can understand the impact of the change.
  - [ ] Improvement (improves and existing functionality): This documentation is needed if the improvement changes the way the user interacts with the system or the outcome of a process/task changes. If it is just, for instance, a performance improvement, documentation might not be needed. 
- [ ] The changed/added documentation is in the project wiki (not privately-hosted pdf files or links pointing to a company website) and is complete and self-explanatory.

### CODE REVIEW

**Comment by norbertbede** [comment]
the name part "createPricelist" is came from our internal ticketing. this is generic issue, pls ignore it.

File: org.adempiere.base/src/org/compiere/process/SvrProcess.java [hunk 1]
```diff
@@ -731,7 +732,7 @@ private void unlock ()
 			MPInstance mpi = new MPInstance (getCtx(), m_pi.getAD_PInstance_ID(), null);
 			if (mpi.get_ID() == 0)
 			{
-				log.log(Level.SEVERE, "Did not find PInstance " + m_pi.getAD_PInstance_ID());
+				log.log(Level.INFO, "Did not find PInstance " + m_pi.getAD_PInstance_ID());
 				return;
 			}
 			mpi.setIsProcessing(false);
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/process/SvrProcess.java:738]: this whole block should take a similar approach as above, i.e check that m_pi.getAD_PInstance_ID() > 0.

**Review by hengsin — CHANGES_REQUESTED** [review]

File: org.adempiere.base/src/org/compiere/process/SvrProcess.java [hunk 2]
```diff
@@ -727,20 +728,17 @@ private void unlock ()
 			//clear interrupt signal so that we can unlock the ad_pinstance record
 			if (Thread.currentThread().isInterrupted())
 				Thread.interrupted();
+			
+			if(m_pi.getAD_PInstance_ID() > 0) {
+				MPInstance mpi = new MPInstance (getCtx(), m_pi.getAD_PInstance_ID(), null);
```
Comment (hengsin) [inline org.adempiere.base/src/org/compiere/process/SvrProcess.java:733]: Should keep the original if(mpi.get_ID() ==0) check here.

Comment (PeterTakacs300) [reply org.adempiere.base/src/org/compiere/process/SvrProcess.java:733]: Can `mpi.get_ID() == 0` if `m_pi.getAD_PInstance_ID() > 0` ?

Comment (PeterTakacs300) [reply org.adempiere.base/src/org/compiere/process/SvrProcess.java:733]: Ok, reverted

**Review by hengsin — APPROVED** [review]
looks good

**Pull request merged**